File: C++/Searching/linear_search.cpp
#include<bits/stdc++.h>
using namespace std;

int linear_search(int arr[],int n,int key){
    for(int i=0;i<n;i++){
        if(arr[i]==key){
            return i;
        }
    }return -1;
    
}
int main()
{
    int n;
    cout<<"Enter the size of array u want :";
    cin>>n;

    int arr[10];
    cout<<"The array u want to enter :";
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }

    int key;
    cout<<"Enter the value u want to search :";
    cin>>key;

    int index=linear_search(arr,n,key); 
    if(index!=-1){
        cout<<key <<" is present at index "<<index<<endl;
    }
    else{
        cout<<key <<" is not present at any index "<<endl;
    }
    return 0;
}

File: Java/linearsearch.java
import java.util.*;
public class linearsearch {
    public static void main(String[] args) {
        Scanner in =new Scanner (System.in);
        int n=in.nextInt();
        int arr[]=new int[n];
        for (int i=0;i<n;i++){
            arr[i]=in.nextInt();
        }
        System.out.println("Enter the target element");
        int target=in.nextInt();
        //search number of frequency of target element at what position
        System.out.println("The target element is present at positions ");
        for (int i=0;i<n;i++){
            if (arr[i]==target){
                System.out.println(i+1);
            }
        }
        //search number of frequency of target element
        int count=0;
        for (int i=0;i<n;i++){
            if (arr[i]==target){
                count++;
            }
        }
        System.out.println("The target element is present "+count+" times");
    }
}
